homeassistant/components/homekit/type_fans.py

```diff
@@ -83,6 +98,56 @@ def set_oscillating(self, value):
                   ATTR_OSCILLATING: oscillating}
         self.call_service(DOMAIN, SERVICE_OSCILLATE, params, oscillating)
 
+    def set_speed_list(self, speed_list):
+        """Set and validate the speed list from the fan."""
+        if self.speed_list == speed_list:
+            return
+        if not speed_list:
+            self.speed_list = []
+            return
+        if SPEED_OFF != speed_list[0]:
+            _LOGGER.warning(
+                "%s: %s (%s) does not contain the speed setting "
+                "%s as its first element. "
+                "Assuming that %s is equivalent to 'off'.""",
+                self.entity_id, ATTR_SPEED_LIST, speed_list,
+                SPEED_OFF, speed_list[0])
+        self.speed_list = speed_list
+
+    def speed_from_percentage(self, percentage):
```

Comment:
`speed_from_percentage` and `percentage_from_speed` should be moved to `homekit/util.py`.
What do you think of:
 - `speed_to_homekit`
 - `speed_to_states`

That's the naming schema we used for temperature values.

Reply:
Okay, good catch that this should be part of a utility. However, if we do this, maybe we should think ahead: 
home-assistant/architecture#27 mentions that other secondary platforms most often expect percentages, therefore I think it would also fit into either `/util/fan_speed.py` or into `/components/fan/util.py`. When doing this, then `*_homekit` would be a bit misleading.

I think of a small class that contains the mapping defined in https://github.com/home-assistant/home-assistant/pull/19767#discussion_r245417379 and that contains has the following methods: `__init__(speed_list: List[Any])`, `as_speed(percentage: int) -> Any` and `as_percent(speed: Any) -> int`. Alternatively, there can be a free-floating function that returns the mapping data structure and 2 such functions for the conversion, but on the other hand this is what classes are for - to group data and associated functions.